zetaclient/chains/solana/repo/repo.go

```diff
@@ -1,54 +1,77 @@
-package rpc
+package repo
 
 import (
 	"context"
 	"strings"
 	"time"
 
-	"github.com/gagliardetto/solana-go"
-	"github.com/gagliardetto/solana-go/rpc"
+	sol "github.com/gagliardetto/solana-go"
+	solrpc "github.com/gagliardetto/solana-go/rpc"
 	"github.com/pkg/errors"
-
-	"github.com/zeta-chain/node/zetaclient/chains/interfaces"
 )
 
 const (
+	// TODO: this does not need to be exported, and can remove pageLimit from GetFirst... params.
+	//
 	// defaultPageLimit is the default number of signatures to fetch in one GetSignaturesForAddressWithOpts call
 	DefaultPageLimit = 1000
 
-	// RPCAlertLatency is the default threshold for RPC latency to be considered unhealthy and trigger an alert.
-	// The 'HEALTH_CHECK_SLOT_DISTANCE' is default to 150 slots, which is 150 * 0.4s = 60s
-	RPCAlertLatency = time.Duration(60) * time.Second
-
 	// see: https://github.com/solana-labs/solana/blob/master/rpc/src/rpc.rs#L7276
 	errorCodeUnsupportedTransactionVersion = "-32015"
 )
 
 // ErrUnsupportedTxVersion is returned when the transaction version is not supported by zetaclient
 var ErrUnsupportedTxVersion = errors.New("unsupported tx version")
 
+type SolanaClient interface {
+	GetSlot(context.Context, solrpc.CommitmentType) (uint64, error)
+
+	GetBlockTime(_ context.Context, block uint64) (*sol.UnixTimeSeconds, error)
+
+	GetAccountInfo(context.Context, sol.PublicKey) (*solrpc.GetAccountInfoResult, error)
+
+	GetTransaction(context.Context,
+		sol.Signature,
+		*solrpc.GetTransactionOpts,
+	) (*solrpc.GetTransactionResult, error)
+
+	GetSignaturesForAddressWithOpts(context.Context,
+		sol.PublicKey,
+		*solrpc.GetSignaturesForAddressOpts,
+	) ([]*solrpc.TransactionSignature, error)
+}
+
+// TODO: This is the start of a modularized repository for Solana, many function in the
```

Comment:
Same, as per the comment above, if this would be included in the issue to create, we can reference the issue link whenever there would be a change for this refactoring